phoenix-core/src/main/java/org/apache/phoenix/query/PhoenixStatsCacheLoader.java

```diff
@@ -43,15 +46,20 @@
     public PhoenixStatsCacheLoader(PhoenixStatsLoader statsLoader, Configuration config) {
         this.statsLoader = statsLoader;
 
-        if (this.executor == null) {
+        if (executor == null) {
             synchronized (PhoenixStatsCacheLoader.class) {
-                if (this.executor == null) {
+                if (executor == null) {
                     // The size of the thread pool used for refreshing cached table stats
                     final int statsCacheThreadPoolSize = config.getInt(
                             QueryServices.STATS_CACHE_THREAD_POOL_SIZE,
                             QueryServicesOptions.DEFAULT_STATS_CACHE_THREAD_POOL_SIZE);
-
-                    this.executor = Executors.newFixedThreadPool(statsCacheThreadPoolSize);
+                    final ThreadFactory threadFactory =
+                            new ThreadFactoryBuilder()
+                                    .setDaemon(true)
+                                    .setNameFormat("PHOENIX-STATS-CACHE-LOADER" + "-thread-%s")
```

Comment:
nit: why not `"PHOENIX-STATS-CACHE-LOADER-thread-%s"` directly?

Reply:
Because that's how it was in the method that I copy-pasted this from :)
I'm going to fix it before commit.